Removes model objects

janakerman opened the pull request:
#37 

Horrible change. No typing. Feels... wrong. Why?
Whyyyy?

Tests are currently broken because the model objects
previously had equals logic to simply compare IDs. Now
we're returning the raw JSON we'll need to do that equality
checking somewhere else.

janakerman committed:
Removes model objects

Horrible change. No typing. Feels... wrong. Why?
Whyyyy?

Tests are currently broken because the model objects
previously had equals logic to simply compare IDs. Now
we're returning the raw JSON we'll need to do that equality
checking somewhere else.

janakerman commented:
Pushed this up so that change is visible and can be discussed. @bjedrzejewski 

I don't like this at all - so little typing - too dynamic for my liking. Can't we all just Java?

janakerman commented:
also fixes #39

janakerman committed:
Fixes unit tests for LichessDataService

Since we weren't parsing to our model objects anymore
we want another way of validating the data (we just check that
id isn't null).

The lichess API just returns whatever and it's no up to the data
service to validate.

Tests needed to be updated to use closures for matching rather
than relying on our model objects .equals() implementation.

bjedrzejewski commented on src/main/java/com/chessmates/controller/EntityController.groovy:
```diff
@@ -22,12 +20,12 @@ class EntityController {
      */
     @GetMapping(value = 'players')
-    List<Player> getPlayers() { entityService.getPlayers() }
+    List getPlayers() { entityService.getPlayers() }
```
Why did you decide to drop the Generic here completely? Can't we have something more detailed than Object? We at least know that this is Json I think?

bjedrzejewski replied:
Ok, I see, thi sis not the way in Groovy. Perhaps maybe we should  be clear that this is actually getPlayersJson() method?

bjedrzejewski commented on src/main/java/com/chessmates/controller/EntityController.groovy:
```diff
@@ -22,12 +20,12 @@ class EntityController {
      */
     @GetMapping(value = 'games')
-    List<Game> getGames() { entityService.getGames() }
+    List getGames() { entityService.getGames() }
```
Same here maybe- getGamesJson- what do you think?

janakerman replied:
As discussed it's not really JSON - I'll change the LichessApi to be more explicit about returning a page.

bjedrzejewski commented on src/main/java/com/chessmates/lichess/data/LichessApi.groovy:
```diff
@@ -11,15 +7,15 @@ interface LichessApi {
      * Get games of a given player.
      */
-    LichessResultPage<Game> getGames(String playerId, int pageNumber, int pageSize)
+    def getGames(String playerId, int pageNumber, int pageSize)
```
LichessResultPage as a wrapper is not a bad idea, this is a wrapper that contains multiple GamesJson I think. We could just abstract that GamesJson as an Object, but shape of the ResultPage could be defined?

janakerman replied:
Agreed - this needs to be renamed to make it clear that we're returning a page of data here, to differentiate it from the getGames methods that return a list of games.

bjedrzejewski commented on src/main/java/com/chessmates/lichess/data/LichessDataServiceImpl.groovy:
```diff
@@ -47,20 +44,25 @@ class LichessDataServiceImpl implements LichessDataService {
-        players.each { playerRepository.save it }
+        List players = resultSet.get()
+                .stream()
+                .filter(LichessEntityValidator.&isValid)
```
Nice 👍

bjedrzejewski commented on src/main/java/com/chessmates/repository/MetaDataRepository.groovy:
```diff
@@ -1,27 +1,24 @@
      * ImmutableMap is returned to indicate that changes to this store don't affect the store contents.
      */
-    ImmutableMap<ImmutablePair<Player, Player>, Game> getLatestGames()
+    ImmutableMap getLatestGames()
```
I don't think we had to lose ImmutablePair to Object mapping here. I think keeping it is a good idea in the signature.

bjedrzejewski commented on src/main/java/com/chessmates/repository/MockGameRepository.groovy:
```diff
@@ -13,22 +11,22 @@ import org.springframework.stereotype.Repository
 
     @Override
-    Game find(String gameId) {
+    Object find(String gameId) {
```
We are not consistent here. Sometimes we use Object, sometimes we leave it empty. It is a super minor comment.

bjedrzejewski reviewed (approved):
Good job, I have a few comments and suggestions that you might think about to improve the general readability here.

janakerman committed:
Responds to code review feedback

- Consistently uses def (or lack of)
- Renames LichessApi methods to be clearer RE pages of data
- Only remove typing for JSON objects - things we control should
still be typed on our class methods.

janakerman committed:
Scheduled tasks no long run in unit tests.

spring-boot-starter-actuator enables scheduling application wide.
This cause me headaches here as I was looking at having a different
configuration for our unit tests that simply turned off scheduling.

Instead, I've used a conditional inclusion on the scheduling bean itself.
It's enabled in the main applicaiton properties and disabled in the test's.

I've addeda comment to the enable scheduling annotation so I hope that
helps future me, or future team members.
